src/nvim/ex_docmd.c

```diff
@@ -7040,30 +7038,33 @@ void ex_cd(exarg_T *eap)
       new_dir = NameBuff;
     }
 #endif
-    if (vim_chdir(new_dir)) {
-      EMSG(_(e_failed));
-    } else {
-      CdScope scope = kCdScopeGlobal;  // Depends on command invoked
+    CdScope scope = kCdScopeGlobal;  // Depends on command invoked
 
-      switch (eap->cmdidx) {
-      case CMD_tcd:
-      case CMD_tchdir:
-        scope = kCdScopeTab;
-        break;
-      case CMD_lcd:
-      case CMD_lchdir:
-        scope = kCdScopeWindow;
-        break;
-      default:
-        break;
-      }
+    switch (eap->cmdidx) {
+    case CMD_tcd:
+    case CMD_tchdir:
+      scope = kCdScopeTab;
+      break;
+    case CMD_lcd:
+    case CMD_lchdir:
+      scope = kCdScopeWindow;
+      break;
+    default:
+      break;
+    }
 
-      post_chdir(scope);
+    if (vim_chdir(new_dir, scope)) {
+      EMSG(_(e_failed));
+      return;
```

Comment:
Won't this cause a memory leak? the `xfree(tofree)` below will be missed. Better to keep the relevant part in an else{} block.